polkadot/runtime/common/src/paras_registrar/mod.rs

```diff
@@ -357,12 +373,29 @@ pub mod pallet {
 		/// Remove a manager lock from a para. This will allow the manager of a
 		/// previously locked para to deregister or swap a para without using governance.
 		///
-		/// Can only be called by the Root origin or the parachain.
+		/// Can be called by the Root origin or the parachain or exceptionally the manager if a parathread.
 		#[pallet::call_index(4)]
-		#[pallet::weight(T::DbWeight::get().reads_writes(1, 1))]
-		pub fn remove_lock(origin: OriginFor<T>, para: ParaId) -> DispatchResult {
-			Self::ensure_root_or_para(origin, para)?;
-			<Self as Registrar>::remove_lock(para);
+		#[pallet::weight(T::DbWeight::get().reads_writes(3, 1))]
+		pub fn remove_lock(origin: OriginFor<T>, id: ParaId) -> DispatchResult {
+			// Ensure that the extrinisic is executed by one of the three allowed origin types
+			match Self::ensure_root_or_para_or_owner(origin.clone(), id) {
+				Ok(ParaOriginType::Root) | Ok(ParaOriginType::Para) => {
+					// The origin is root or para, no action needed
+				}
+				Ok(ParaOriginType::Manager) => {
+					// The origin is the para manager, check if the chain is a parathread
+					let id_lifecycle = paras::Pallet::<T>::lifecycle(id).ok_or(Error::<T>::NotRegistered)?;
+					if id_lifecycle != ParaLifecycle::Parathread {
+						return Err(Error::<T>::NotParathread.into());
+					}
```

Comment:
In the near future, with Coretime all chains will be "LifeCycle::Parathread". Basically this lifecycle stuff gets removed any way and this is not a good idea to introduce this change here.

Reply:
Then this should be part of a greater review of the `para_registrar` functionality at that time - because there are some complications implied by having only one lifecycle type not just for the lock, but for swapping also. 

As for existing parachains, as far as I know, have no way to force a downgrade to become a parathread during their lease periods.  In that case they have to live out their lifecycles which could still be nearly two years from now and what would happen to the locked funds in that case?

Ok though, assuming that "all chains will be 'LifeCycle::Parathread'" then the swap mechanism becomes irrelevant, and the para manager role has little sense either since it will have no meaningful utility - which is at odds with putting up funds. Which is why I think this at least has some utility in the interim - simply because there is a lot to refactor and it will become technical debt in the end.

Reply:
@bkchr Are you suggesting I remove the code you've selected from the PR? I can of course do that and adjust the unit test, but if I do - then this will permit the para owner to unlock without restriction. Is this the intention?